source/NVDAObjects/IAccessible/winword.py

```diff
@@ -265,15 +266,35 @@ def script_reportCurrentHeaders(self,gesture):
 		columnText=self.fetchAssociatedHeaderCellText(cell,True)
 		ui.message("Row %s, column %s"%(rowText or "empty",columnText or "empty"))
 
-	def script_caret_moveByCell(self,gesture):
+	def script_caret_moveByCell(self, gesture: inputCore.InputGesture) -> None:
+		info = self.makeTextInfo(textInfos.POSITION_SELECTION)
+		inTable = info._rangeObj.tables.count > 0
+		if not inTable:
+			gesture.send()
+			return
+		oldSelection = info.start, info.end
 		gesture.send()
-		info=self.makeTextInfo(textInfos.POSITION_SELECTION)
-		inTable=info._rangeObj.tables.count>0
-		isCollapsed=info.isCollapsed
-		if inTable:
-			info.expand(textInfos.UNIT_CELL)
-			speech.speakTextInfo(info, reason=controlTypes.OutputReason.FOCUS)
-			braille.handler.handleCaretMove(self)
+		start = time.time()
+		retryInterval = 0.01
+		maxTimeout = 0.15
+		elapsed = 0
+		while True:
+			if scriptHandler.isScriptWaiting():
+				# Prevent lag if keys are pressed rapidly
+				return
+			info = self.makeTextInfo(textInfos.POSITION_SELECTION)
+			newSelection = info.start, info.end
+			if newSelection != oldSelection:
+				log.debug(f"Detected new selection after {elapsed} sec")

```

Comment:
Ensure elapsed is up to date when logging

```suggestion
			if newSelection != oldSelection:
				elapsed = time.time() - start
				log.debug(f"Detected new selection after {elapsed} sec")
```